Create API documentation per source in docs/api and summary with link…

pedritom-amzn opened the pull request:
…s in README.md.

notes:
* the new covidcast documentation is a little behind the latest changes, but i plan to update in a follow-up commit.
* this change does not update the project's root README. That is also left for a follow-up commit.

undefx commented on docs/api/README.md:
```diff
@@ -0,0 +1,212 @@
+| [`quidel`](quidel.md) | Quidel | Data provided by Quidel Corp., which contains flu lab test results. | yes |
+| [`sensors`](sensors.md) | Delphi's Digital Surveillance Sensors | ... <!-- TODO --> | no |
+| [`twitter`](twitter.md) | Twitter Stream | Estimate of influenza activity based on analysis of language used in tweets. | yes |
+| [`wiki`](wiki.md) | Wikipedia Access Logs | Number of page visits for selected English, Influenza-related wikipedia articles. | ... <!-- TODO --> |
```
fyi this is unrestricted

pedritom-amzn replied:
replaced TODO with "no"

undefx commented on docs/api/README.md:
```diff
@@ -0,0 +1,212 @@
+| [`nowcast`](nowcast.md) | ILI Nearby | A nowcast of U.S. national, regional, and state-level (weighted) percent ILI, available seven days (regionally) or five days (state-level) before the first ILINet report for the corresponding week. | no |
+| [`quidel`](quidel.md) | Quidel | Data provided by Quidel Corp., which contains flu lab test results. | yes |
+| [`sensors`](sensors.md) | Delphi's Digital Surveillance Sensors | ... <!-- TODO --> | no |
+| [`twitter`](twitter.md) | Twitter Stream | Estimate of influenza activity based on analysis of language used in tweets. | yes |
```
we should prominently acknowledge http://HealthTweets.org/

pedritom-amzn replied:
added link to description.

undefx commented on docs/api/README.md:
```diff
@@ -0,0 +1,212 @@
+# Example URLs
+
+### FluView on 2015w01 (national)
+https://delphi.midas.cmu.edu/epidata/api.php?source=fluview&regions=nat&epiweeks=201501
```
currently broken, host should be delphi.midas.**cs**.cmu.edu

pedritom-amzn replied:
did a replace-all in the docs/api directory.

undefx commented on docs/api/README.md:
```diff
@@ -0,0 +1,212 @@
+
+### Wiki article "influenza" on 2015w01
+
+https://delphi.midas.cmu.edu/epidata/api.php?source=wiki&articles=influenza&epiweeks=201501
```
currently broken, host should be delphi.midas.**cs**.cmu.edu

pedritom-amzn replied:
fixed.

undefx commented on docs/api/README.md:
`````diff
@@ -0,0 +1,212 @@
+cat(paste(res$result, res$message, length(res$epidata), "\n"))
+````
+
+# Related work
+
+ - Cook, Samantha, et al. "Assessing Google flu trends performance in the United States during the 2009 influenza virus A (H1N1) pandemic." PloS one 6.8 (2011): e23610.
+ - Broniatowski, David A., Michael J. Paul, and Mark Dredze. "National and local influenza surveillance through Twitter: an analysis of the 2012-2013 influenza epidemic." (2013): e83672.
+ - Dredze, Mark, et al. "HealthTweets. org: A Platform for Public Health Surveillance using Twitter." AAAI Conference on Artificial Intelligence. 2014.
+ - Generous, Nicholas, et al. "Global disease monitoring and forecasting with Wikipedia." (2014): e1003892.
+ - Hickmann, Kyle S., et al. "Forecasting the 2013–2014 Influenza Season Using Wikipedia." (2015): e1004239.
+ - McIver, David J., and John S. Brownstein. "Wikipedia usage estimates prevalence of influenza-like illness in the United States in near real-time." PLoS Comput Biol 10.4 (2014): e1003581.
`````
im realizing that this really needs to be updated, but i think i'll have to let delphi handle that

undefx commented on docs/api/afhsb.md:
```diff
@@ -0,0 +1,72 @@
+| --- | --- | --- |
+| `auth` | password | string |
+| `epiweeks` | epiweeks | `list` of epiweeks |
+| `locations` | locations | `list` of [region](labels/regions.txt), [state](labels/states.txt), or 3-letter country code labels |
```
here, similar ones in other docs: these relative links are broken (i think) now that this is in the `docs/` dir

pedritom-amzn replied:
did a replace-all for "(labels/" -> "(../../labels/" and "(src/" -> "(../../src/".

undefx commented on docs/api/covidcast.md:
```diff
@@ -0,0 +1,108 @@
+
+Any data which is produced novelly by Delphi and is intentionally and openly
+surfaced by Delphi through this API is hereby licensed
+[CC-BY](https://creativecommons.org/licenses/by/4.0/). The `covid_survey_county_weekly` endpoint
```
covid_survey_county_weekly -> covidcast

pedritom-amzn replied:
fixed (and a couple other locations)

undefx commented on docs/api/dengue_nowcast.md:
```diff
@@ -0,0 +1,72 @@
+
+Any data which is produced novelly by Delphi and is intentionally and openly
+surfaced by Delphi through this API is hereby licensed
+[CC-BY](https://creativecommons.org/licenses/by/4.0/). The `delphi` endpoint is known to wholly or partially serve data
```
looks like copypasta from `delphi` endpoint, recommend removing

pedritom-amzn replied:
replaced with the non-delphi licensing stub i use elsewhere.

pedritom-amzn replied:
actually, the reason i concluded this was a delphi set and used the delphi licensing wording (albeit with the incorrect source name) was because in places like the following it is referred to as delphi's:
https://github.com/cmu-delphi/delphi-epidata/blob/c48af8f126e935306eaebb79fcf1b22a201a3dfa/src/client/delphi_epidata.py#L475

similar for the dengue_sensors set. was this an incorrect conclusion?

undefx commented on docs/api/dengue_sensors.md:
```diff
@@ -0,0 +1,72 @@
+
+Any data which is produced novelly by Delphi and is intentionally and openly
+surfaced by Delphi through this API is hereby licensed
+[CC-BY](https://creativecommons.org/licenses/by/4.0/). The `delphi` endpoint is known to wholly or partially serve data
```
same here (not `delphi`)

pedritom-amzn replied:
changed to `dengue_sensors`.

pedritom-amzn replied:
note, i left the delphi licensing language because of comments like the following in the code:

https://github.com/cmu-delphi/delphi-epidata/blob/c48af8f126e935306eaebb79fcf1b22a201a3dfa/src/client/delphi_epidata.py#L509

i can switch to the non-delphi licensing language stub if this was an incorrect conclusion.

undefx commented on docs/api/meta.md:
```diff
@@ -0,0 +1,165 @@
+
+## Data licensing
+
+The data surfaced through this API is just a carefully curated mirror of data acquired from .... <!-- TODO -->
```
this one is actually delphi's own original source

pedritom-amzn replied:
changed to the delphi licensing wording.

undefx commented on docs/api/nidss_dengue.md:
```diff
@@ -0,0 +1,146 @@
+This is the documentation of the API for accessing the NIDSS Dengue (`nidss_dengue`) data source of
+the [Delphi](https://delphi.cmu.edu/)'s epidemiological data.
+
+## FluView Clinical Data
```
fluview -> nidss

pedritom-amzn replied:
fixed

undefx commented on docs/api/nidss_flu.md:
```diff
@@ -0,0 +1,167 @@
+This is the documentation of the API for accessing the NIDSS Flu (`nidss_flu`) data source of
+the [Delphi](https://delphi.cmu.edu/)'s epidemiological data.
+
+## FluView Clinical Data
```
fluview -> nidss

pedritom-amzn replied:
fixed

undefx commented on src/server/api_helpers.php:
```diff
@@ -100,7 +100,7 @@ function filter_dates($field, $dates) {
 // builds a SQL expression to filter values/ranges of integers (ex: epiweeks)
 //   $field: name of the field to filter
-//   $epiweeks: array of integer values/ranges
+//   $values: array of integer values/ranges
```
nice catch!

undefx commented:
i'll plan to merge this tomorrow. want to give time for any additional comments.

pedritom-amzn commented:
thanks for catching all those copy-pasta errors! i believe i've addressed all the feedback except for the refactoring suggestions which i'll leave for a later change. please note that i've put questions around whether the `dengue_{nowcast,sensors}` sources are delphi data or not.

undefx commented:
thanks again!
